### PR TITLE
Added an offensive spells threshold. 

### PR DESCRIPTION
**Pull request by jayRyan24**
Added the ability to stop training offensive spells at a defined threshold. Suggested by Binu.

### CODE REVIEW

File: profiles/base.yaml [hunk 1]
```diff
@@ -115,6 +115,7 @@ dump_junk: false
 dump_item_count: 9
 use_barb_combos: false
 whirlwind_trainables:
+offensive_spell_training_max_threshold:
```
Comment (dr-leustyin) [inline profiles/base.yaml]: I think this should default to 32, then line 1304 in combat-trainer could be simplified

File: combat-trainer.lic [hunk 2]
```diff
@@ -1298,7 +1301,9 @@ class SpellProcess
   def check_offensive(game_state)
     return if game_state.casting
     return if game_state.npcs.empty?
+    return if @offensive_spells_max_threshold && game_state.npcs.length > @offensive_spells_max_threshold
```
Comment (dr-leustyin) [inline combat-trainer.lic]: game_state.npcs.length should be magic mindstates

File: profiles/base.yaml (hunk 1 above)
Comment (jayRyan24) [reply profiles/base.yaml]: I just copied the other implementation. I agree but followed suit.

File: combat-trainer.lic (hunk 2 above)
Comment (jayRyan24) [reply combat-trainer.lic]: I completely misread the other threshold, wtf.

**Pull request merged**